SOFTENG310-G5/BrainSomething
# Refactor/refactor project structure

## Context

This PR addresses the need to optimize the project structure by removing unnecessary files from version control, renaming folders for consistency, and organizing styles into a dedicated folder. These changes improve maintainability and follow best practices in file structure and project organization.

Closes #69 

## What Changed?

- Removed `node_modules` from the repository and added it to `.gitignore`.
- Renamed the `Backend` folder to `backend` to follow lowercase naming conventions.
- Moved all scattered CSS files into a central `styles/` folder.
- Updated all import paths in JavaScript files to point to the new CSS file locations in `styles/`.

## How To Review

1. Start by reviewing the changes in `.gitignore` to ensure the proper exclusion of `node_modules`.
2. Check the folder renaming from `Backend` to `backend`.
3. Review the `styles/` folder to confirm that all CSS files were moved there.
4. Verify that all CSS imports in relevant components and pages have been updated correctly.
5. Check for any unnecessary files or changes.

## Testing

- After the refactor, tested the project to ensure all components and pages are working as expected with the updated CSS imports.
- Verified that no styles are broken and all functionality remains intact.
- Confirmed that no `node_modules` files are committed and everything rebuilds correctly after running `npm install`.

## Risks

- Import path issues: The main risk lies in missing or incorrect paths when updating the CSS imports. Focus on verifying that all paths to the CSS files are correct in the JS files.
- Folder renaming issues: Some operating systems may have issues with renaming folders where only the casing changes (e.g., `Backend` to `backend`), so double-check to ensure the folder is renamed correctly without conflicts.

## Notes

- Be sure to reinstall dependencies locally with `npm install` after the `node_modules` removal.
- Thorough testing has been done to ensure the refactor does not introduce style or functionality issues.
